File: pythonPoker/playPoker.py
from array import *
from operator import itemgetter, attrgetter
import sys
import random

class Card:
    def __init__(self, face, suit):
        self.face=face
        self.suit=suit

    def __repr__(self):
        f=self.face
        s=self.suit
        if self.face== 1:
            f="A"
        elif self.face== 11:
            f="J"
        elif self.face== 12:
            f="Q"
        elif self.face==13:
            f="K"
           
        if self.suit==1:
            s="S"
        elif self.suit==2:
            s="H"
        elif self.suit==3:
            s="C"
        elif self.suit==4:
            s="D"
    
        return f"{f}{s}"

class Player:
    def __init__(self):
        #self.name = name
        self.hand = []

    def draw(self, deck):
        card = deck.draw()
        if card:
            self.hand.append(card)
        else:
            print("The deck is empty.")

   # def rankHand(self):
    #    self.hand.sort(key=lambda card: card.face)
        #temp.showHand()
      
class Deck:
    def __init__(self):
        self.suits = ['S', 'H', 'C', 'D']
        self.values = [1,2,3,4,5,6,7,8,9,10,11,12,13]
        self.cards = [{'value': value, 'suit': suit} for suit in self.suits for value in self.values]

    def shuffle(self):
        random.shuffle(self.cards)

    def draw(self):
        if not self.is_empty():
            return self.cards.pop()
        else:
            return None

    def is_empty(self):
        return len(self.cards) == 0

    def printDeck(self):
        for c in self.cards:
            print(c.values+""+c.suits)



##################################################################################################







#p.rankHand()






arguments= sys.argv



if len(arguments)==2:
    print(arguments[1])



else:

    d=Deck()
    d.shuffle()
    d.printDeck()

    #Table=[Player(), Player(), Player(), Player(), Player(), Player()]
    

File: juliaPoker/playPoker.jl
using Random

mutable struct Deck
    cards::Vector{Tuple{Int, Int}}
    
    function Deck()
        suits = 1:4
        faces = 1:13
        cards = [(suit, face) for suit in suits for face in faces]
        new(cards)
    end
    
end
    
function deal(deck::Deck)
    if length(deck.cards) > 0
        return pop!(deck.cards)
    else
        return nothing  # Return nothing if the deck is empty
    end
end

function showDeck(deck::Deck)
    index = 1
    for card in deck.cards
        print(card_to_string(card), " ")
        if index == 13
            println()
            index = 1
        else
            index += 1
        end
    end
end

function shuffle(deck:: Deck)
    Random.shuffle!(deck.cards)
end

function card_to_string(card::Tuple{Int, Int})
    suits = ["S", "H", "C", "D"]
    faces = string.([2,3,4,5,6,7,8,9,10, "J", "Q", "K", "A"])
    suit_str = suits[card[1]]
    face_str = faces[card[2]]
    return "$face_str$suit_str"
end


#  players= [Player(), Player(), Player()]